src/MICore/Debugger.cs

```diff
@@ -578,20 +578,23 @@ public Task CmdBreak(BreakRequest request)
             return CmdBreakInternal();
         }
 
-        internal bool IsLocalGdb()
+        protected bool IsLocalLaunchUsingServer()
         {
-            return (this.MICommandFactory.Mode == MIMode.Gdb &&
-               this._launchOptions is LocalLaunchOptions &&
-               String.IsNullOrEmpty(((LocalLaunchOptions)this._launchOptions).MIDebuggerServerAddress));
+            return (_launchOptions is LocalLaunchOptions &&
+                (!String.IsNullOrWhiteSpace(((LocalLaunchOptions)_launchOptions).MIDebuggerServerAddress) ||
+                 !String.IsNullOrWhiteSpace(((LocalLaunchOptions)_launchOptions).DebugServer)));
+        }
 
+        internal bool IsLocalGdbTarget()
+        {
+            return (MICommandFactory.Mode == MIMode.Gdb && !IsLocalLaunchUsingServer());
```

Comment:
Unless I am missing something, this will give a different result than the old `IsLocalGdb` function would give if `this._launchOptions is LocalLaunchOptions` is false.

Reply:
In other words, I think you want to change this to `MICommandFactory.Mode == MIMode.Gdb && this._launchOptions is LocalLaunchOptions && !IsLocalLaunchUsingServer()`

Reply:
Ummm, yes and no, and hence the change in the method name. The old `IsLocalGdn()` also wanted to make sure that we are not using `MIDebuggerServerAddress` meaning a server. So does this one but using it also makes sure that `DebugServer` is not being used. _And that was actually the bug_. It was never looking for a 'is this a purely locally running gdb?`. I looked at every place `IsLocalGdb()` and `IsRemoteGdb()` is/was used and what we really needed to know was if there is a server in play or gdb controlling the target directly.

Reply:
Yup, I got that, but now this function will incorrectly return true in a different case - if we aren't dealing with local launch options.

Reply:
Yes, I see that now.

Reply:
Done. It now has the true meaning of what `IsLocalGdbTarget()` should mean. Did another round of testing and pushed changes.

Reply:
Just to acknowledge, what @gregg-miskelly pointed out was a serious defect